src/components/DataTable/__tests__/DataTable-test.js

```diff
@@ -303,4 +308,201 @@ describe('DataTable', () => {
+            fieldB: 'Field 3:B',
+          },
+        ],
+        headers: [
+          {
+            key: 'fieldA',
+            header: 'Field A',
+          },
+          {
+            key: 'fieldB',
+            header: 'Field B',
+          },
+        ],
+        locale: 'en',
+        render: jest.fn(
+          ({
+            rows,
+            headers,
+            getHeaderProps,
+            getSelectionProps,
+            getBatchActionProps,
+            getRowProps,
+            onInputChange,
+          }) => (
+            <TableContainer title="container">
+              <TableToolbar>
+                <TableBatchActions {...getBatchActionProps()}>
+                  <TableBatchAction onClick={jest.fn()}>Ghost</TableBatchAction>
+                </TableBatchActions>
+                <TableToolbarSearch onChange={onInputChange} />
+              </TableToolbar>
+              <Table>
+                <TableHead>
+                  <TableRow>
+                    <TableExpandHeader />
+                    <TableSelectAll {...getSelectionProps()} />
+                    {headers.map(header => (
+                      <TableHeader {...getHeaderProps({ header })}>
+                        {header.header}
+                      </TableHeader>
+                    ))}
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  {rows.map(row => (
+                    <React.Fragment key={row.id}>
+                      <TableExpandRow {...getRowProps({ row })}>
+                        <TableSelectRow {...getSelectionProps({ row })} />
+                        {row.cells.map(cell => (
+                          <TableCell key={cell.id}>{cell.value}</TableCell>
+                        ))}
+                      </TableExpandRow>
+                      {row.isExpanded && (
+                        <TableExpandedRow>
+                          <TableCell colSpan={headers.length + 3}>
+                            <h1>Expandable row content</h1>
+                            <p>Description here</p>
+                          </TableCell>
+                        </TableExpandedRow>
+                      )}
+                    </React.Fragment>
+                  ))}
+                </TableBody>
+              </Table>
+            </TableContainer>
+          )
+        ),
+      };
+    });
+
+    it('should add additional rows when receiving new props', () => {
+      const wrapper = mount(<DataTable {...mockProps} />);
+      const args = mockProps.render.mock.calls[0][0];
+
+      expect(args.rows.length).toEqual(mockProps.rows.length);
+
+      const nextRows = [
+        ...mockProps.rows,
+        {
+          id: 'd',
+          fieldA: 'Field 4:A',
+          fieldB: 'Field 4:B',
+        },
+      ];
+
+      wrapper.setProps({ rows: nextRows });
+
+      const nextArgs = mockProps.render.mock.calls[1][0];
+      expect(nextArgs.rows.length).toBe(nextRows.length);
```

Comment:
@joshblack Are you updating test for row ID order here...? Thanks!

Reply:
Just added!